<!DOCTYPE html>
<html lang="en">
    <head>
        <title>Im Lab's PredictDB Data Repository</title>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="author" content="Jiamao Zheng, Hae Kyung Im">
        <meta name="description" content="PredictDB">
        <link rel="shortcut icon" href="http:/aws.amazon.com/favicon.ico">
        <link rel="stylesheet" href="http://maxcdn.bootstrapcdn.com/bootstrap/3.2.0/css/bootstrap.min.css">
        <link rel="stylesheet" href="http://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.1.0/css/font-awesome.min.css">
        <link rel="stylesheet" href="http://cdn.datatables.net/plug-ins/725b2a2115b/integration/bootstrap/3/dataTables.bootstrap.css">
        <link rel="stylesheet" href="/css/predictdb.css">
        <script src="http://code.jquery.com/jquery-2.1.1.min.js"></script>
        <script src="http://maxcdn.bootstrapcdn.com/bootstrap/3.2.0/js/bootstrap.min.js"></script>
        <script src="http://cdnjs.cloudflare.com/ajax/libs/bootbox.js/4.2.0/bootbox.min.js"></script>
        <script src="http://sdk.amazonaws.com/js/aws-sdk-2.0.13.min.js"></script>
        <script src="https://sdk.amazonaws.com/js/aws-sdk-2.5.0.min.js"></script>
        <script src="http://cdnjs.cloudflare.com/ajax/libs/moment.js/2.7.0/moment.min.js"></script>
        <script src="http://cdn.datatables.net/1.10.2/js/jquery.dataTables.min.js"></script>
        <script src="http://cdn.datatables.net/plug-ins/725b2a2115b/integration/bootstrap/3/dataTables.bootstrap.js"></script>
        <script type="text/javascript" src="/js/s3.js"> </script>
    </head>

    <body>
    <nav class="navbar navbar-inverse navbar-fixed-top" id="Nav">
        <div class="container">
          <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
              <span class="sr-only">Toggle navigation</span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="http://predictdb.org"><strong>PredictDB</strong></a>
          </div>
          <div class="collapse navbar-collapse btn-group" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav">
                <li><a href="http://predictdb.org"><strong>Home</strong></a></li>
                <li><a href="/News.html"><strong>News</strong></a></li>
                <li><a href="/FAQ.html"><strong>FAQ</strong></a></li>
                <li><a href="/Contact.html"><strong>Contact Us</strong></a></li>
            </ul>
           </div>
         </div>
        </nav>

<div class="container">
    <br />
    <br />
    <br />

<div class="container">
    <div class="page-header">
        <h1><strong> FAQ </strong><small>Frequently Asked Questions</small></h1>
    </div>

    <div class="panel-group" id="accordion">
        <div class="faqHeader">General questions</div>
            <div class="panel panel-default">
                <div class="panel-heading">
                    <h4 class="panel-title">
                        <a class="accordion-toggle collapsed" data-toggle="collapse" data-parent="#accordion" href="#collapseThree">Q: How was the expression data preprocessed for training the prediction models?</a>
                    </h4>
                </div>
                <div id="collapseThree" class="panel-collapse collapse">
                    <div class="panel-body">
                        A: We used the expression pre-processed by the GTEx analysis working group. Covariates are the same as used in the eQTL analysis. GTEx processing pipeline is <a href="https://github.com/broadinstitute/gtex-pipeline/" target="_blank"> here </a>, and Docker image is <a href="https://hub.docker.com/r/broadinstitute/gtex_eqtl/" target="_blank"> here </a>. A copy of <strong> Portal_Analysis_Methods_v6p_08182016.pdf </strong> can be downloaded from either <a href="https://storage.googleapis.com/gtex-public-data/Portal_Analysis_Methods_v6p_08182016.pdf" target="_blank"> GTEx </a> or <a href="https://s3.amazonaws.com/predictdb/More/Portal_Analysis_Methods_v6p_08182016.pdf" target="_blank"> our predictdb s3 bucket</a>.              
                    </div>
                </div>
            </div>
        <div class="faqHeader">PredictDB questions </div>
        <div class="panel panel-default">
            <div class="panel-heading">
                <h4 class="panel-title">
                    <a class="accordion-toggle collapsed" data-toggle="collapse" data-parent="#accordion" href="#collapseTwo">Q: Why are there fewer gene models in the latest release?</a>
                </h4>
            </div>
            <div id="collapseTwo" class="panel-collapse collapse">
                <div class="panel-body">
                A: To reduce the likelihood spurious findings, we are now filtering models based on statistical significance (FDR < .05).                
                </div>
            </div>
        </div>
        <div class="panel panel-default">
            <div class="panel-heading">
                <h4 class="panel-title">
                    <a class="accordion-toggle collapsed" data-toggle="collapse" data-parent="#accordion" href="#collapseFour">Q: Can I predict for genes on chromosome X or Y?</a>
                </h4>
            </div>
            <div id="collapseFour" class="panel-collapse collapse">
                <div class="panel-body">
                A: Not at the moment.  Sex chromosome genes need to be handled differently than those on the autosomes, and do not fit into our existing analysis pipeline.                
                </div>
            </div>
        </div>
        <div class="panel panel-default">
            <div class="panel-heading">
                <h4 class="panel-title">
                    <a class="accordion-toggle collapsed" data-toggle="collapse" data-parent="#accordion" href="#collapseFive">Q: How was the covariance data calculated?</a>
                </h4>
            </div>
            <div id="collapseFive" class="panel-collapse collapse">
                <div class="panel-body">
                A: For the GTEx models, we calculated the covariance of the SNP dosages within our training samples.  For the DGN model, we used public genotype data from 1000 Genomes to calculate the covariance of the SNP dosages.                
                </div>
            </div>
        </div>
        <div class="panel panel-default">
            <div class="panel-heading">
                <h4 class="panel-title">
                    <a class="accordion-toggle collapsed" data-toggle="collapse" data-parent="#accordion" href="#collapseSix">Q: What does “TW_” mean in the file names?</a>
                </h4>
            </div>
            <div id="collapseSix" class="panel-collapse collapse">
                <div class="panel-body">
                A: This is an abbreviation for “Tissue-Wide.”  We have historically created a Cross-Tissue model, which is a measure of the expression common across all tissues, and Tissue-Specific models (TS), which is modeled as an orthogonal component to the shared cross tissue component. Details can be found <a href="http://dx.doi.org/10.1101/043653" target="_blank"> here </a>. While these models are not available at the moment, they are in our pipeline.             
                </div>
            </div>
        </div>
        <div class="panel panel-default">
            <div class="panel-heading">
                <h4 class="panel-title">
                    <a class="accordion-toggle collapsed" data-toggle="collapse" data-parent="#accordion" href="#collapseSeven">Q: Are performance measures of prediction available? </a>
                </h4>
            </div>
            <div id="collapseSeven" class="panel-collapse collapse">
                <div class="panel-body">
                A: Yes, they can be found in the extra table in each sqlite db.             
                </div>
            </div>
        </div>
        <div class="panel panel-default">
            <div class="panel-heading">
                <h4 class="panel-title">
                    <a class="accordion-toggle collapsed" data-toggle="collapse" data-parent="#accordion" href="#collapseEight">Q: How many samples were used to train each model? </a>
                </h4>
            </div>
            <div id="collapseEight" class="panel-collapse collapse">
                <div class="panel-body">
                A: This can be found in the <b> sample_info </b> table in each sqlite db.              
                </div>
            </div>
        </div>
        <div class="panel panel-default">
            <div class="panel-heading">
                <h4 class="panel-title">
                    <a class="accordion-toggle collapsed" data-toggle="collapse" data-parent="#accordion" href="#collapseNine">Q: How do I query the prediction model db? </a>
                </h4>
            </div>
            <div id="collapseNine" class="panel-collapse collapse">
                <div class="panel-body">
                A: Find example queries in <a href="https://github.com/hakyimlab/PrediXcan/blob/master/Software/query-db.Rmd" target="_blank"> link </a>             
                </div>
            </div>
        </div>
        <div class="panel panel-default">
            <div class="panel-heading">
                <h4 class="panel-title">
                    <a class="accordion-toggle collapsed" data-toggle="collapse" data-parent="#accordion" href="#collapseTen">Q: Which models should I use: 1KG or HapMap? And is there a list of SNPs used for the models? </a>
                </h4>
            </div>
            <div id="collapseTen" class="panel-collapse collapse">
                <div class="panel-body">
                <p> 
                    A: <b>1KG</b> and <b>HapMap</b> are the labels for the set of SNPs which were used to train the prediction models.  As you probably have guessed, <b>1KG</b> refers to the 1000 Genomes SNP set and the <b>HapMap</b> refers to the HapMap SNP set.  Careful consideration of the SNPs in your genotype data should help you decide which of the models to use.  The 1000 Genomes models make use of many more SNPs, so we believe these models to have greater prediction accuracy and some preliminary investigation into out-of-sample data suggests this to be the case.  If you are working with older data though, or you would have to impute a significant amount of your genotype data to achieve more complete coverage of the models, we recommend using the HapMap models.  For most use cases, the HapMap models are a good starting point.
                </p>
                <p> 
                    Click on the following links to download the complete SNP annotation files we used to train the different models.  These are tab-delimited text files with info on chromosome, position and allele info in addition to the rsid numbers. <br>
                    <a href="https://s3.amazonaws.com/imlab-open/Data/PredictDB/GTEx_hapmap_snps.annot.txt.gz" target="_blank"> HapMap (36MB) </a> <br>
                    <a href="https://s3.amazonaws.com/imlab-open/Data/PredictDB/GTEx_OMNI_genot_1KG_imputed_var_info4_maf01_CR95_CHR_POSb37_ID_REF_ALT_release_v6.txt.gz" target="_blank"> 1000 Genomes (150MB) </a> <br>
                    <!-- <a href="https://s3.amazonaws.com/imlab-open/Data/PredictDB/hapmapSnpsCEU.tar.gz" target="_blank"> DGN HapMap (61MB) </a> -->
                </p>
                <p> 
                    Alternatively, you can query the sqlite database to get a list of all the SNPs it uses for prediction.  See <a href="https://gist.github.com/ScottPDickinson/40534d396b97d7281094fab32baca92a" target="_blank"> here </a> for a python function to create a file containing all SNPs in a database.    
                </p>       
                </div>
            </div>

        </div>

        <div class="panel panel-default">
            <div class="panel-heading">
                <h4 class="panel-title">
                    <a class="accordion-toggle collapsed" data-toggle="collapse" data-parent="#accordion" href="#collapse11">Q: How can I build my own prediction model? </a>
                </h4>
            </div>
            <div id="collapse11" class="panel-collapse collapse">
                <div class="panel-body">
                <p> 
                    A: All scripts for model training are located here: https://github.com/hakyimlab/PredictDBPipeline
Note Keep in mind that large chunk of the scripts have to do with splitting up the data and submitting to the HPC cluster at the University of Chicago.  The scripts you would probably be most interested in for model training and building a database are <a href="https://github.com/hakyimlab/PredictDBPipeline/blob/master/scripts/GTEx_Tissue_Wide_CV_elasticNet.R" target="_blank"> GTEx_Tissue_Wide_CV_elasticNet.R </a> and <a href="https://github.com/hakyimlab/PredictDBPipeline/blob/master/scripts/make_sqlite_db.py" target="_blank"> make_sqlite_db.py </a>
                </p>    
                </div>
            </div>
            
        </div>
        <div class="faqHeader">PrediXcan questions </div>
        <div class="panel panel-default">
            <div class="panel-heading">
                <h4 class="panel-title">
                    <a class="accordion-toggle collapsed" data-toggle="collapse" data-parent="#accordion" href="#collapseA1">Q: PrediXcan says my samples file has too few/too many rows. </a>
                </h4>
            </div>
            <div id="collapseA1" class="panel-collapse collapse">
                <div class="panel-body">
                   A: In order for the predicted expression file to be correct, the dosage files and samples file must correspond to the same individual. The columns for the dosage files are columns are snpid rsid position allele1 allele2 MAF id1 ..... idn and the samples file lists out the information for id1 ... idn row by row. As the id numbers are not included in the dosage file, it is critically important that the samples file has the same number of individuals in the same order as the dosages files. Otherwise later association tests will be invalid.
                </div>
            </div>
        </div>
        <div class="faqHeader">MetaXcan questions</div>
    </div>
</div>

<style>
    .faqHeader {
        font-size: 27px;
        margin: 20px;
    }

    .panel-heading [data-toggle="collapse"]:after {
        font-family: 'Glyphicons Halflings';
        content: "\e072"; /* "play" icon */
        float: right;
        color: #F58723;
        font-size: 18px;
        line-height: 22px;
        /* rotate "play" icon from > (right arrow) to down arrow */
        -webkit-transform: rotate(-90deg);
        -moz-transform: rotate(-90deg);
        -ms-transform: rotate(-90deg);
        -o-transform: rotate(-90deg);
        transform: rotate(-90deg);
    }

    .panel-heading [data-toggle="collapse"].collapsed:after {
        /* rotate "play" icon from > (right arrow) to ^ (up arrow) */
        -webkit-transform: rotate(90deg);
        -moz-transform: rotate(90deg);
        -ms-transform: rotate(90deg);
        -o-transform: rotate(90deg);
        transform: rotate(90deg);
        color: #454444;
    }
</style>

<!-- Bootstrap FAQ - END -->

</div>
         <br><br><br>
        <div class="footer" role="contentinfo" id="footer">
        <div class="container">
            <br>
            <p class="copyright text-muted"> <center> <a href="http://hakyimlab.org" style="color: #FFFFFF">Im Lab</a>, Genetic Medicine, Department of Medicine,  The University of Chicago. <a href="#" style="color: #FFFFFF">Back to top</a> </center> </p>
        </div>
        <div class="counter">
            <script type="text/javascript">
            var sc_project=11075605; 
            var sc_invisible=0; 
            var sc_security="3fe3753b"; 
            var scJsHost = (("https:" == document.location.protocol) ?
            "https://secure." : "http://www.");
            document.write("<sc"+"ript type='text/javascript' src='" + scJsHost+
            "statcounter.com/counter/counter.js'></"+"script>");
            </script>
            <noscript><div class="statcounter"><a title="website statistics"
            href="http://statcounter.com/free-web-stats/" target="_blank"><img
            class="statcounter" src="//c.statcounter.com/11075605/0/3fe3753b/0/"
            alt="website statistics"></a></div></noscript>
        </div>
        <br>
      </div>
  </body>
</html>

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-61894206-2', 'auto');
  ga('send', 'pageview');

</script>